### PR TITLE
Session status tracking

### PR DESCRIPTION
**Pull request by olehkorkh-planeks**

### CODE REVIEW

**Review by suhlrich — CHANGES_REQUESTED** [review]
Looks good. One change requested.

File: mcserver/views.py [hunk 1]
```diff
@@ -1310,7 +1310,72 @@ def neutral_img(self, request, pk):
...
+            serializer = SessionIdSerializer(Session.objects.none(), many=True)
+            if filtering_serializer.is_valid():
+                status_str = filtering_serializer.validated_data.get('status')
+                date_range = filtering_serializer.validated_data.get('date_range')
+                filter_kwargs = {'status': status_str}
+                if date_range:
+                    filter_kwargs['status_changed__gte'] = date_range[0]
+                    filter_kwargs['status_changed__lte'] = date_range[1]
+                if not IsAdmin().has_permission(request, self) and not IsBackend().has_permission(request, self):
+                    filter_kwargs['user'] = request.user
```
[the remark this answers is not in the document]
Comment (olehkorkh-planeks) [reply mcserver/views.py:1329]: @suhlrich I added the commit. Now an optional argument `username` is supported.
https://github.com/stanfordnmbl/opencap-api/pull/156/commits/9a3ca4d6f7f4650ff187778ad7fbfb6daa612b8d

**Review by suhlrich — APPROVED** [review]

**Comment by suhlrich** [comment]
Do we need to migrate now? @antoinefalisse

**Comment by antoinefalisse** [comment]
there is a bunch of stuff in dev, will do soon

**Pull request merged**